libs/jsonschema/instance.go

```diff
@@ -0,0 +1,90 @@
+package jsonschema
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+// Load a JSON document, also validating it against the JSON schema. Instance here
+// refers to a JSON document. see: https://json-schema.org/draft/2020-12/json-schema-core.html#name-instance
+func (s *Schema) LoadInstance(path string) (map[string]any, error) {
+	instance := make(map[string]any, 0)
+	b, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(b, &instance)
+	if err != nil {
+		return nil, err
+	}
+
+	// The default JSON unmarshaler parses untyped number values as float64.
+	// We convert integer properties from float64 to int64 here.
+	for name, v := range instance {
+		propertySchema, ok := s.Properties[name]
+		if !ok {
+			continue
+		}
+		if propertySchema.Type != IntegerType {
+			continue
+		}
+		integerValue, err := toInteger(v)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse property %s: %w", name, err)
+		}
+		instance[name] = integerValue
+	}
+	return instance, s.ValidateInstance(instance)
+}
+
+func (s *Schema) ValidateInstance(instance map[string]any) error {
+	if err := s.validateAdditionalProperties(instance); err != nil {
+		return err
+	}
+	if err := s.validateRequired(instance); err != nil {
+		return err
+	}
+	return s.validateTypes(instance)
+}
+
+// If additional properties is set to false, this function validates instance only
+// contains properties defined in the schema.
+func (s *Schema) validateAdditionalProperties(instance map[string]any) error {
+	if s.AdditionalProperties != false {
```

Comment:
AdditionalProperties has the type `any`. This is because the JSON schema spec allows you to define a schema for all additional properties. The is not false assertion also guards against the case when it's nil / a `*Schema` object.

Reply:
Thanks for clarifying.